app/src/main/java/in/avimarine/boatangels/general/Versioning.java

```diff
@@ -0,0 +1,58 @@
+package in.avimarine.boatangels.general;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.util.Log;
+import com.google.firebase.firestore.DocumentSnapshot;
+import in.avimarine.boatangels.db.FireBase;
+import in.avimarine.boatangels.db.iDb;
+
+/**
+ * Avi Marine Innovations - www.avimarine.in
+ *
+ * Created by Amit Y. on 19/03/2016.
+ */
+public class Versioning {
+
+  private static final String TAG = "Versioning";
+  private Context c;
+  private iDb commManager;
+
+  public Versioning(Context c) {
+    this.c = c;
+    commManager = new FireBase();
+  }
+
+  public void getSupportedVersion(OnGetSupportedVersionListener<Long> listener) {
+    try {
+      commManager.getSupportedVersion(task -> {
+        if (task.isSuccessful()) {
+          DocumentSnapshot document = task.getResult();
+          long supportedVersion;
+          if (!document.exists()) {
+            Log.d(TAG,"No supported version value found in DB");
+            supportedVersion = getInstalledVersion();
+            commManager.setSupportedVersion(supportedVersion);
+            Log.d(TAG,"Setting the minimal supported version to current version: " + supportedVersion);
+          } else {
+            supportedVersion = document.getLong("compatibleVersion");
+          }
+          listener.onComplete(supportedVersion);
+        }
+      });
+    } catch (Exception e) {
+      Log.e(TAG, "error getting supported version", e);
+    }
+  }
+
+  public int getInstalledVersion() {
+    try {
+      PackageInfo pInfo = c.getPackageManager().getPackageInfo(c.getPackageName(), 0);
+      return pInfo.versionCode;
+    } catch (PackageManager.NameNotFoundException e) {
+      Log.d(TAG, "Error retreiving versioncode: ", e);
+    }
+    return -1;
```

Comment:
Make sure there's no way to have a lower number of version saved in the DB, else, write a handling of this case.

Reply:
A very remote possibility, but I added a fix.